### PR TITLE
set pkg/apis/monitoring tag to v0.42.1

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/prometheus-operator/prometheus-operator/issues/3530

### CODE REVIEW

**Comment by lilic** [comment]
Thanks for the PR, I actually have not yet finished the 0.42.1 release by bringing it into master, but this is a good start! Thanks so much! Do you mind just updating the release notes here with the above steps, its not needed just would be nice. If not I can do it in another PR as well! https://github.com/prometheus-operator/prometheus-operator/blob/master/RELEASE.md

**Comment by TheoBrigitte** [comment]
You're welcome. Yes sure I can do that, you mean updating the CHANGELOG.md right ?

**Comment by lilic** [comment]
Sorry was not clear enough, the CHANGELOG I can update in a follow up PR, I meant adding the change you just did in go.mod file to the RELEASE.md process so we don't forget this step in the future, hope that makes more sense?

**Comment by TheoBrigitte** [comment]
Ha ok understood :+1:

**Comment by TheoBrigitte** [comment]
Wait we can't actually do this, because the tag is not yet there when creating the release. So `go build` will fail since it won't be able to find the new tag.

**Comment by TheoBrigitte** [comment]
`pkg/apis/monitoring` will always be 1 version behind in go.mod then. Feels weird to me. Otherwise this has to happen after the release was created.

**Comment by simonpasquier** [comment]
Yes we need to tag `github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring` before cutting the prometheus-operator release.

**Comment by TheoBrigitte** [comment]
This should do it:

* from `master` set the new tag on `pkg/apis/monitoring`
* move to release branch and prepare the release including getting the new `pkg/apis/monitoring` version
* do the release

Tag from `pkg/apis/monitoring` will always be behind, but I do not see another way of achieving this.

File: RELEASE.md [hunk 1]
```diff
@@ -31,16 +31,31 @@ Maintaining the release branches for older minor releases happens on a best effo
...
+A number of files have to be re-generated, this is automated with the following make target:
 
 ```bash
 $ make clean generate
 ```
 
+Bump the version for `pkg/apis/monitoring` package in go.mod:
+
+```bash
+$ go get -u github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring
```
Comment (simonpasquier) [inline RELEASE.md:56]:
```suggestion
$ go get "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring@v${VERSION}"
```

**Comment by TheoBrigitte** [comment]
Are we good here?

**Comment by brancz** [comment]
Thank you!

**Pull request merged**